### ISSUE TITLE
[Android] No banner shown

### ISSUE THREAD

**Issue by fokoz** [evollu/react-native-fcm#638]
Even if I use `custom_notification` in the data key (without notification key) it shows only a tiny icon at left corner and in the system tray, no rectangle banner shown.

1. What version of RN and react-native-fcm are you running? 10.0.2
2. What device are you using? (e.g iOS9 emulator, Android 6 device)? Android 6
3. Is your app running in foreground, background or not running? All

**Comment by evollu** [comment]
set priority: 'high'

**Comment by fokoz** [comment]
Here is my payload:
```
"data": {
                "content_available": true,
                "custom_notification": {
                    "title": "Title",
                    "body": "Body",
                    "badge": 1,
                    "color": "#00ACD4",
                    "priority" : "high",
                    "show_in_foreground": true
                }
            }
```
Did I miss something such as notification icon banner or something about permission ?

**Comment by evollu** [comment]
the banner should show up even with icon.
`"content_available": true,` should be outside of data object.

does local notification work?

**Comment by fokoz** [comment]
Local notification also has no banner shown

**Comment by fokoz** [comment]
@evollu 
I have another queston ( if I have to open another issue, I will)

I managed to resolve this issue https://github.com/evollu/react-native-fcm/issues/637
And when I passed `custom_notification` payload, I found this bug instead https://github.com/wix/react-native-navigation/issues/2130.

But if i passed just `notification` payload, it is fine.

My question is, what does `custom_notification` actually do ?

**Comment by muzammil-triffort** [comment]
@fokoz Have you found any solution for this? I am unable to fix this for Android.

**Comment by ducpt2** [comment]
I have same issue, i don't know how to use custom_notification, when i use example, don't know where to get API KEY in FirebaseConstants.
const FirebaseConstants = {
	KEY: "YOUR_API_KEY"
}

Anyone know how to do it? Thanks.

**Comment by nighttiger1990** [comment]
@ducpt2 : https://console.firebase.google.com/u/0/ 
1. Create New Project
2. Go to Settings
3. Click tab Cloud Messaging
4. Copy Server Key or Legacy Server Key (both is Key API you need)

**Comment by ducpt2** [comment]
@nighttiger1990 thanks, 
i already found it. 
I'm trying to show banner when app in background but it can't work, only work when app foreground. Any help?

**Comment by muzammil-triffort** [comment]
@ducpt2 I am also having the same issue :( Please post your flow if you find something. Thanks in advance.

**Comment by ducpt2** [comment]
@muzammil-triffort try to change code of your server like document, it will work. your server need to check ios, android to send data to fcm. Remember, android need 'custom_notification', and it will work.

**Comment by muzammil-triffort** [comment]
@ducpt2 Thank you for your reply. For testing, I am using http://pushtry.com/ for Android FCM. But no success. Below is my sample code. i also tried to place priority inside notification dictionary. 
`{
"to":"dVYPAwCTkzQ:APA91bFAiy1C2t0a7s4r0Yl1VYAx4",
"data":
{
"priority":"high"
"show_in_foreground": true,
"custom_notification":
	{
"title":"Simple FCM Client",
	"body":"This is a notification with only NOTIFICATION.",
	"sound":"default",
}
},
}`

**Comment by ducpt2** [comment]
@muzammil-triffort you should use sample code in Example folder, i build that and test, and it work like a charm. If it not, you should check your setting of android. Good luck.

**Comment by muzammil-triffort** [comment]
You must be talking about this sample:
https://github.com/evollu/react-native-fcm/tree/master/Examples/simple-fcm-client

Is sample code is working for Android also? Because i was getting some compilation error.

**Comment by ducpt2** [comment]
@muzammil-triffort Yes, it not so difficult, you can copy code to new project, get Token and hardcode it, it like a server, you can call to fcm to pushNotification. :)

**Comment by muzammil-triffort** [comment]
Okay Thanks. I will try.

**Comment by diegorodriguesvieira** [comment]
The same issue here? Someone found the fix? =(

**Comment by ducpt2** [comment]
@diegorodriguesvieira i have already said it :)

**Comment by mymai91** [comment]
@ducpt2 Can you get notification in the background App?
